src/components/credit-card-date/__tests__/creditcard-test.js

```diff
@@ -0,0 +1,25 @@
+/** @jsx React.DOM */
+
+var CreditCardDateView = require('../code/views/creditCardDateView.jsx');
+
+describe('CreditCardDateComponent', function() {
+
+  it('is an element', function() {
+    var creditCardDate = TestUtils.renderIntoDocument(
+      <CreditCardDateView />
+    );
+
+    var renderedCreditCardDate = TestUtils.findRenderedDOMComponentWithClass(creditCardDate, 'component-credit-card-date');
+    assert.isDefined(renderedCreditCardDate);
+  });
+
+  it('should render a label', function() {
+    var creditCardDate = TestUtils.renderIntoDocument(
+      <CreditCardDateView label="Expires: " />
+    );
+
+    var renderedCreditCardLabel = TestUtils.findRenderedDOMComponentWithClass(creditCardDate, 'credit-card-date-label');
+    assert.isDefined(renderedCreditCardLabel);
+  });
+
+});
```

Comment:
can we test:
- not having a label
- having an error
- not having an error
- defining custom values for attributes
- checking custom attributes
- event handlers
...

I'm not sure if it's possible to test all of the above and more, but there seems to be a bit of functionality in this component that isn't currently tested. I feel that since these components are starting to go into production we need to vastly increase our test coverage and a new component seems like the perfect place to start. Using [`React.Addons.TestUtils`](https://facebook.github.io/react/docs/test-utils.html#simulate) we should be able to simulate each of these circumstances.